### DISCUSSION TITLE
Type check errors in v7.0

### DISCUSSION THREAD

**Discussion by davidbrochart** [fastapi-users/fastapi-users#719] [Q&A] [answered]
Using fastapiusers 7.0.0, mypy type checking leads to the following errors:
```
error: Definition of "is_verified" in base class "BaseUser" is incompatible with definition in base class "BaseUserUpdate"
error: Definition of "is_superuser" in base class "BaseUser" is incompatible with definition in base class "BaseUserUpdate"
error: Definition of "email" in base class "BaseUser" is incompatible with definition in base class "BaseUserUpdate"
error: Definition of "is_active" in base class "BaseUser" is incompatible with definition in base class "BaseUserUpdate"
```
In https://github.com/fastapi-users/fastapi-users/commit/83873338f3cb3f10c6886abe51b946e830a7eb9c, `is_verified`, `is_superuser`, `is_active` and `email` were added as `Optional` in `BaseUserUpdate`, while they already exist in `CreateUpdateDictModel` from which it derives. Maybe they don't need to be present again in `BaseUserUpdate`?

**Comment by frankie567** [comment, marked as answer]
Oh yeah, it may be a change I overlooked: does your `UserUpdate` still inherit from `User`? If so, you should remove the inheritance and repeat your custom fields in the `UserUpdate` model.

If that's the issue, I'll update the release notes.

**Reply by davidbrochart** [reply]
> does your `UserUpdate` still inherit from `User`?

Yes.

> If so, you should remove the inheritance and repeat your custom fields in the `UserUpdate` model.

That worked, thanks a lot!

**Reply by frankie567** [reply]
Thank you for raising this! I'll update the release notes!
